src/common/types.ts

```diff
@@ -34,6 +51,12 @@ export type RegisterResourceFunction = (
     cb: ReadResourceTemplateCallback
 ) => RegisteredResourceTemplate;
 
+export type RegisterPromptFunction = <Args extends ZodRawShape>(name: string, config: {
+  title?: string;
+  description?: string;
+  argsSchema?: Args;
+}, cb: PromptCallback<Args>) => RegisteredPrompt;
```

Comment:
Just double checking, are all fields genuinely optional?

Reply:
i've taken this type declaration from one defined in the typescript-mcp sdk but not exported out.